internal/format/middle_visitor.go

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package format
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/emicklei/proto"
+	"github.com/uber/prototool/internal/settings"
+	"github.com/uber/prototool/internal/text"
+)
+
+type middleVisitor struct {
```

Comment:
Seeing as there are only two visitors, what are your thoughts on renaming these types to `preVisitor` and `postVisitor`, or something similar? We could leave it as `firstVisitor` and rename this to `lastVisitor`?

Or, depending on if we add context to these types (mentioned above), maybe a more descriptive name for these types will come out of it.

Reply:
Yea I was going to have a visitor at the end, just never renamed. How about `firstPassVisitor` and `mainVisitor`?

Reply:
sgtm

Reply:
Done